### ISSUE TITLE
http://ci.solvcon.net/job/examples.vewave_cvg/ fails around revision 51e85bf107a3a85b9930b4cabc5f3aa13b9c59bf

### ISSUE THREAD

**Issue by yungyuc** [solvcon/pastsolvcon#93]
Originally reported by: **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**

---

BB: https://bitbucket.org/solvcon/solvcon/commits/51e85bf107a3/

---
- Bitbucket: https://bitbucket.org/solvcon/solvcon/issue/93

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

I don't quite understand this issue.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Please see http://ci.solvcon.net/job/examples.vewave_cvg/40/.  With your changes around 51e85bf107a3a85b9930b4cabc5f3aa13b9c59bf, the example/answer test at $scroot/examples/vewave/cvg/ failed.  There is an error introduced in your changes to cause that fail.  Could you fix the error(s)?

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

I will look into it.  Those files are not I usual work with.  I might accidently change it.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Thanks.  It looks like they are affected by your change to solvcon.parcel.vewave.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Hi @ericpo , do you have time on this issue?  Do you need help on it?

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

I am sorry that I am busy this week so I don't have to look into it yet.  I will do it this week.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

Hi, I just found out that I did not commit this.  It seems that you commit in this commit
https://bitbucket.org/ericpo/solvcon_vewave/commits/a18102476a2917ebdd6518267981f19e4937b19c

It seems this commit merges in that when I merge my solvcon_vewave with solvcon.
I'm not sure what I should do now.  Do you want it back to original or keep it?  Thanks.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Nice catch!  It did fail once at http://ci.solvcon.net/job/examples.vewave_cvg/37/, but after I reran the test, it passed: http://ci.solvcon.net/job/examples.vewave_cvg/38/.  From the history I guess a181024 isn't the killer.

But maybe you've done some tests and are sure that the change a181024 is the guilty?  Have you tried to back out the change in your local build and tested it locally?  Could you elaborate more about how you investigate this issue?  Thanks.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

No I have not done any test.  I check your error message, #40 has same error as #37, and they seem to relate to multiprocessing and rpc.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

You are right that http://ci.solvcon.net/job/examples.vewave_cvg/40/console failed because of RPC error in unit tests, like http://ci.solvcon.net/job/examples.vewave_cvg/37/console (URL is easier to be clicked).  But in the rerun http://ci.solvcon.net/job/examples.vewave_cvg/41/console (notice the associated revision, which is the same as http://ci.solvcon.net/job/examples.vewave_cvg/40/console), the unit tests passed and then the error changed.  After that, most failures showed the same error message (I didn't check all, but the 47th run is):

```
#!
+ examples/vewave/run_vewave_cvg.sh
/home/ychen/opt/scruntime/bin/python2.7 go run cvg2d_200
Traceback (most recent call last):
  File "go", line 281, in <module>
    solvcon.go()
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/cmdutil.py", line 187, in go
    cmd()
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/command.py", line 728, in __call__
    func = arrangements[name]
KeyError: 'cvg2d_200'
```

I can reproduce this error by simply running the following command locally:

```
#!
$ examples/vewave/run_vewave_cvg.sh
/Users/yungyuc/opt/anaconda/bin/python2.7 go run cvg2d_200
Traceback (most recent call last):
  File "go", line 281, in <module>
    solvcon.go()
  File "/Users/yungyuc/work/coding/solvcon/dist/SOLVCON-0.1.2+/solvcon/cmdutil.py", line 187, in go
    cmd()
  File "/Users/yungyuc/work/coding/solvcon/dist/SOLVCON-0.1.2+/solvcon/command.py", line 728, in __call__
    func = arrangements[name]
KeyError: 'cvg2d_200'
```

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

Are you running from example or sandbox.  Either does not have cvg2d_200 arrangement.  I think maybe you can run sandbox first.  It seems that I mess up the arrangement in example.  I will work on that.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

The vewave_cvg answer test runs from the examples/ directory.  Please see the configuration of http://ci.solvcon.net/job/examples.vewave_cvg/ and the file $root/examples/vewave/run_vewave_cvg.sh.

The test case is created to provide a minimal regression for the in-development vewave solver.  Its failure means something trivial is changed.  @ericpo, I think we should look into it and fix it.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

As I mentioned, there is no cvg2d_200 in the example.  I will fix it.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

http://ci.solvcon.net/job/examples.vewave_cvg/49/console

```
#!

+ examples/vewave/run_vewave_cvg.sh
/home/ychen/opt/scruntime/bin/python2.7 go run cvg2d_0_0
Traceback (most recent call last):
  File "go", line 248, in <module>
    sc.go()
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/cmdutil.py", line 187, in go
    cmd()
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/command.py", line 751, in __call__
    func(submit=False, **funckw)
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/case.py", line 901, in simu
    cse = func(*args, **kw)
  File "go", line 242, in cvg2d_0_0
    rho=1.06e3, vp=1578.0, sig0=10.0, freq=2e6)
  File "go", line 226, in cvg2d_ve
    return cvg_base(casename=casename, **kw)
  File "go", line 195, in cvg_base
    mesher(casename, bcmap)
  File "go", line 120, in mesher
    save_blk(blk, meshname)
  File "go", line 61, in save_blk
    bio.save(stream=meshname)
  File "/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/dist/SOLVCON-0.1.2+/solvcon/io/block.py", line 792, in save
    stream = open(stream, 'wb')
IOError: [Errno 2] No such file or directory: '/home/ychen/var/lib/jenkins/workspace/examples.vewave_cvg/examples/vewave/cvg/mesh/vewave2d_0.blk'
```

It didn't work.  It looks like a os.makedirs() should have been there.  I think you may need to test it locally with all generated files removed in your local clone.

@ericpo , I am granting you the write/push permission to this repository, so that you can directly push to this repository and Jenkins will build.  I think in this way you can fix this more efficiently.

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

It seemed that I've granted you write permission to you long ago, @ericpo .  You can push now.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

Ok.  I will keep testing.  Thanks.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

Hi, I push it but it seems example/vewave did not build.  Did I do something wrong?

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Because the "release" project failed http://ci.solvcon.net/job/release/54/console.  It's a random RPC error.  Currently ci.solvcon.net is configure to only proceed to answer test projects if the "release" test project passes.

As I said in slack, rerun the "release" build should fix it.

**Comment by yungyuc** [comment]
_Original comment by_ **Po-Hsien Lin (BitBucket: [ericpo](http://bitbucket.org/ericpo), GitHub: [ericpo](http://github.com/ericpo))**:

---

It passed! 
http://ci.solvcon.net/job/examples.vewave_cvg/50/

**Comment by yungyuc** [comment]
_Original comment by_ **Yung-Yu Chen (BitBucket: [yungyuc](http://bitbucket.org/yungyuc), GitHub: [yungyuc](http://github.com/yungyuc))**:

---

Fixed by commit 3230255.  Thanks @ericpo.
